### PR TITLE
Create public datatype classes

### PR DESCRIPTION
**Pull request by cel243**
All shared datatypes used throughout our design.

### CODE REVIEW

File: src/main/java/com/google/sps/data/ProjectMetaData.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  /** 
+   *  Create metadata that speicfy that this project had an average of
+   *  {@code averageIAmBindingsInPastYear} IAM Bindings per day in the past 
+   *  year. 
+   */
+  public ProjectMetaData(float averageIAMBindingsInPastYear) {
+    this.averageIAMBindingsInPastYear = averageIAMBindingsInPastYear;
+  }
+}
```
Comment (Ionis-Kutrolli) [inline src/main/java/com/google/sps/data/ProjectMetaData.java:16]: Missing newline at the end of file

Comment (cel243) [reply src/main/java/com/google/sps/data/ProjectMetaData.java:16]: It's not missing---there would be a little red circle with a slash if it were

**Comment by peterpie123** [comment]
LGTM

File: src/main/java/com/google/sps/data/IAMBindingDatabaseEntry.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.google.sps.data;
```
Comment (davidgaskins) [inline src/main/java/com/google/sps/data/IAMBindingDatabaseEntry.java:1]: is this the appropriate package name? would the three of you want to come up with something fun and related to the project title?

File: src/main/java/com/google/sps/data/IAMBindingDatabaseEntry.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+/** Represents a single row of the IAM Bindings table. */
+public class IAMBindingDatabaseEntry {
+
+  public String projectId;
+  public long timestamp;
+  public int numberBindings;
```
Comment (davidgaskins) [inline src/main/java/com/google/sps/data/IAMBindingDatabaseEntry.java:8]: nit: bindingsNumber.

File: src/main/java/com/google/sps/data/IAMRecommenderMetadata.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package com.google.sps.data;
+
+import com.google.sps.data.Recommendation.RecommenderType;
+
+/** Class that provides additional information about IAM Binding recommendations. */
+public class IAMRecommenderMetadata extends RecommenderMetadata {
+
+  public int impactInIAMBindings;
```
Comment (davidgaskins) [inline src/main/java/com/google/sps/data/IAMRecommenderMetadata.java:8]: Could you please provide a link to this impact? E.g. Is it from Recommender API, etc?

File: src/main/java/com/google/sps/data/IAMBindingDatabaseEntry.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package com.google.sps.data;
+
+/** Represents a single row of the IAM Bindings table. */
+public class IAMBindingDatabaseEntry {
+
+  public String projectId;
```
Comment (davidgaskins) [inline src/main/java/com/google/sps/data/IAMBindingDatabaseEntry.java:6]: Is this meant to be mutable?

If not, please consider @AutoValue or at least wrapping controlling the visibility of these members and wrapping them in getters. Here and throughout this PR.

File: src/main/java/com/google/sps/data/Project.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package com.google.sps.data;
+
+/** Represents a project. */
+public class Project {
+
+  public String name;
+  public String id;
+  public String projectNumber;
```
Comment (davidgaskins) [inline src/main/java/com/google/sps/data/Project.java:8]: Should this be a "ProjectIdentification"?

File: src/main/java/com/google/sps/data/IAMBindingDatabaseEntry.java (hunk 2 above)
Comment (cel243) [reply src/main/java/com/google/sps/data/IAMBindingDatabaseEntry.java]: we've renamed it to com.google.impactdashboard

File: src/main/java/com/google/sps/data/Project.java (hunk 6 above)
Comment (cel243) [reply src/main/java/com/google/sps/data/Project.java]: I can wrap it in a projectIdentification object

File: src/main/java/com/google/sps/data/IAMRecommenderMetadata.java (hunk 4 above)
Comment (cel243) [reply src/main/java/com/google/sps/data/IAMRecommenderMetadata.java]: done!

File: src/main/java/com/google/sps/data/IAMBindingDatabaseEntry.java (hunk 5 above)
Comment (cel243) [reply src/main/java/com/google/sps/data/IAMBindingDatabaseEntry.java]: I've changed all the classes to @AutoValue classes so they're all generated now!

**Pull request merged**